### ISSUE TITLE
Identify can't identify jpeg2000 without filename extension

### ISSUE THREAD

**Issue by jcoyne** [ImageMagick/ImageMagick#28]
Identify can identify standard jpegs without a filename hint as to their type:
```
cp fixtures/image.jpg /opt/6789
identify /opt/6789
/opt/6789 JPEG 512x465 512x465+0+0 8-bit sRGB 114KB 0.000u 0:00.000
```
And it can identify jpeg2000s that have a filename extension:
```
identify fixtures/image.jp2
fixtures/image.jp2 JP2 512x465 512x465+0+0 8-bit sRGB 11KB 0.000u 0:00.000
```

But when I remove the extension, it can no longer identify the file:
```
cp fixtures/image.jp2 /opt/12345
identify /opt/12345
identify: no decode delegate for this image format `' @ error/constitute.c/ReadImage/501.
```

```
$ identify --version
Version: ImageMagick 6.9.1-10 Q16 x86_64 2015-08-07 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2015 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules
Delegates (built-in): bzlib freetype jng jp2 jpeg lcms ltdl lzma png tiff xml zlib
```

**Comment by acozine** [comment]
Works with older version: 
```
[alicia@ip-10-157-104-54 circuit-court-records-TEI-textimage]$ identify 0991_008_no_ext 
0991_008_no_ext JP2 1000x644 1000x644+0+0 8-bit DirectClass 704KB 0.280u 0:00.279
[alicia@ip-10-157-104-54 circuit-court-records-TEI-textimage]$ identify --version
Version: ImageMagick 6.7.8-9 2014-06-10 Q16 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2012 ImageMagick Studio LLC
Features: OpenMP    
```

**Comment by jcoyne** [comment]
So, this probably broke with the move from Jasper to openjpeg2

**Comment by dlemstra** [comment]
Thanks for reporting this issue, this will be resolved in the next version of ImageMagick (6.9.2-1)
